server/npg/porch/models/task.py

```diff
@@ -50,3 +50,37 @@ class Task(BaseModel):
 
     def generate_task_id(self):
         return hashlib.sha256(ujson.dumps(self.task_input, sort_keys=True).encode()).hexdigest()
+
+    def __eq__(self, other):
+        '''
+        Allow instances of Task to be compared with ==
+
+        The pipeline and task_input_ids can partially differ and it still be a
+        valid comparison. Clients do not get to create task_input_ids and may
+        not fully specify a pipeline. Status is also optional
+
+        Automatically attempts to cast a dict into a Task, and therefore
+        ignores any properties not valid for a Task
+
+        '''
+        if not isinstance(other, Task):
+            if isinstance(other, dict):
+                other = Task.parse_obj(other)
```

Comment:
Capture an error parsing an object, return False? Or let it error?

Reply:
I think I'd rather it errored